src/routes/two-factor-authentication/disable-secret.ts

```diff
@@ -0,0 +1,26 @@
+import Joi from 'joi';
+import { ControllerGeneratorOptionsWithoutClientOrSupplier } from '../../comms/controller';
+
+interface Request {
+  query: {
+    password: string;
+  },
```

Comment:
Would you mind moving this into the body? See https://stackoverflow.com/a/2629241 for why.